tests/performance-test/README.md

```diff
@@ -3,18 +3,139 @@
 ## Introduction
 
 The performance test provides an automated environment in which to to run stress
-tests on the SAF locally using Minishift. Collectd-tg is used to simulate
-extensive netwrok traffic to pump through SAF. Because Minishift only supports a
-single node at a time, this test demonstrates the limits of SAF in a constrained
-environment. Test scenarios are manually configured in a yaml file and results
-can be analyzed in a series of grafana dashboards.
+tests on SAF. Collectd-tg or telemetry-bench are used to simulate extensive
+metrics data to pump through SAF. Results of testing can be analyzed in a
+grafana dashboard.
 
 Two additional pods are deployed by the performance test: one that hosts a
 grafana instance and one that executes the testing logic.
 
 ![A Performance Test Dashboard](images/dashboard.png)
 
-## Configuring Tests
+## Environment
+
+* openshift v3.11.135
+* docker v17.05+
+
+## Setup
+
+SAF must already be deployed, a quick way to do this is using the
+`quickstart.sh` script in `telemetry-framework/deploy/` directory to run SAF.
+Here is an example of how to do that in minishift:
+
+>NOTE: if you intend to rebuild any test containers, your openshift must have an
+>external route for the registry. The command for that is included here.
+
+```shell
+minishift addons enable registry-route   # Run BEFORE starting minishift
+minishift start
+eval $(minishift oc-env)
+cd $WORKDIR/telemetry-framework/deploy/; ./quickstart.sh
+```
+
+More details about deploying SAF can be found in the
+[SAF deployment docs](../../../docs/README.adoc#quickstart-script-for-development-usage)
+
+## Deploying Grafana
+
+Ensure that all of the SAF pods are already marked running with `oc get pods`.
+Next, launch the grafana instance for test results gathering. This only needs
+to be done once:
+
+```shell
+cd $WORKDIR/telemetry-framework/tests/performance-test/deploy
+./grafana-launcher.sh
+```
+
+The grafana launcher script will output a URL that can be used to log into the
+dashboard. This Grafana instance has all authentication disabled - if, in the
+future, the performance test should report to an authenticated grafana instance,
+the test scripts must be modified.
+
+## Launching the test
+
+Once the Grafana instance is running, launch the performance test OpenShift job:
+
+```shell
+./performance-test.sh
+```
+
+Monitor the performance test status by watching the job with
+`oc get job -l app=saf-performance-test -w`. Logs can be viewed with
+`oc logs saf-perftest-<NUM>-runner-<ID> -f`
+
+## [DEPRECATED] Support for collectd-tg as a test tool
+
+Currently we favor telemetry-bench because it:
+
+* Makes predictable instead of random metrics
+* Can generate exactly the number of metrics we ask for
+* Can scale up to multiple instances by varying the hostname prefix
+* Has a codebase we control directly
+
+Collectd-tg has it's place in some collectd->QDR related testing, so we are
```

Comment:
Could potentially link to the upstream tool?

`[collectd-tg](https://collectd.org/documentation/manpages/collectd-tg.1.shtml)